### ISSUE TITLE
Cannot build app with a dependency that imports its package.json

### ISSUE THREAD

**Issue by m-mujica** [stealjs/steal-tools#490]
Basically, if there is a dependency that loads its package.json like:

```
var pkg = require('./package');
```

You'll get the following error when running steal's build

![screen shot 2016-06-30 at 08 41 34](https://cloud.githubusercontent.com/assets/724877/16488429/df3df93a-3ea6-11e6-8a8a-7dea2708509c.png)

The problem seems to be that the json source it's not being transformed to 

`module.exports = {"name" ...}`

before the minification happens here https://github.com/stealjs/steal-tools/blob/c6f735639ddffe17fed8d35ea591e217c8ab87fa/lib/graph/transpile.js#L59-L61

Example app https://github.com/m-mujica/steal-build-package-json-bug

**Comment by pYr0x** [comment]
don't know why this should work....

if your dummy-pkg module contains
```
var pkg = require('./package.json');
```
all works like expected.
steal/system-json is the extension thats handle `json`.
see tests https://github.com/stealjs/system-json/blob/master/test/test.js

**Comment by m-mujica** [comment]
@pYr0x in Node if you do `require('./package')` it works, we support that to make sure those packages are loaded by steal correctly. See https://github.com/stealjs/system-npm/blob/293a073e4845c612e031593a9acf92dfb0b45fa0/test/import_test.js#L183

**Comment by m-mujica** [comment]
*Update*: I have benn working on this, it seems `load` isn't mutable and [this line](https://github.com/stealjs/system-npm/blob/293a073e4845c612e031593a9acf92dfb0b45fa0/npm-extension.js#L369) isn't working at all.

I've change system-npm to add the keep the updated address in `load.metadata` https://github.com/stealjs/system-npm/tree/retry-build-bug and then I changed system-json to look for `load.metadata.address` or `load.address` https://github.com/stealjs/system-json/tree/check-address.

I made these changes to my example app but I'm getting a different error this time, at some point normalize is being called with an undefined name

<img width="535" alt="screen shot 2016-07-05 at 09 44 29" src="https://cloud.githubusercontent.com/assets/724877/16587375/5f4e6a66-4298-11e6-8216-82331747083f.png">

Will look into this later....

**Comment by matthewp** [comment]
@m-mujica Adding the check for load.address causes it to process your `package.json!npm` module through this path, which we obviously don't want. The fix will be to see if the module is using a plugin, and if so it shouldn't be treated like a non-plugin json module.

**Comment by matthewp** [comment]
Fixed by #530
